File: Day_165/evaluate_reverse_polish_notation.py
def main():
    tokens = ["2", "1", "+", "3", "*"]
    stack = []

    for token in tokens:
        if token not in "+-*/":
            stack.append(int(token))
        else:
            b = stack.pop()
            a = stack.pop()

            if token == "+":
                stack.append(a + b)
            elif token == "-":
                stack.append(a - b)
            elif token == "*":
                stack.append(a * b)
            else:
                stack.append(int(a / b))

main()

File: Day_192/evaluate_reverse_polish_notation.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    vector<string> tokens = {"2", "1", "+", "3", "*"};

    stack<int> st;

    for (string token : tokens) {
        if (token != "+" && token != "-" &&
            token != "*" && token != "/") {
            st.push(stoi(token));
        }
    }

    while (!st.empty()) {
        cout << st.top() << " ";
        st.pop();
    }

    return 0;
}